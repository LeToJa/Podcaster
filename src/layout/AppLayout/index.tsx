import { Outlet } from 'react-router-dom'

import { AppProvider } from '../../context/AppContext/context'

import NavigationHeader from '../NavigationHeader'

const AppLayout = () => <>
    <AppProvider>
        <main className='max-w-screen-xl mx-auto'>
            <NavigationHeader />
            <Outlet />
        </main>
    </AppProvider>
</>

export default AppLayout