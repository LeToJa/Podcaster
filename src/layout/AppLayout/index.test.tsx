import { render } from '@testing-library/react'
import { describe, it, expect, vi } from 'vitest'

import AppLayout from '.'

vi.mock('react-router-dom', () => ({
    ...vi.importActual,
    Outlet: () => <figure data-testid="outlet">Outlet Component</figure>,
}))

vi.mock('../NavigationHeader', () => ({
    default: () => <figure>NavigationHeader</figure>,
}))

describe('AppLayout', () => {
    it('renders correctly and matches snapshot', () => {
        const { container } = render(<AppLayout />)

        expect(container).toMatchSnapshot()
    })

    it('renders the NavigationHeader', () => {
        const { getByText } = render(<AppLayout />)

        expect(getByText('NavigationHeader')).toBeTruthy()
    })

    it('renders the Outlet component', () => {
        const { getByTestId } = render(<AppLayout />)

        expect(getByTestId('outlet')).toBeTruthy()
    })
})
