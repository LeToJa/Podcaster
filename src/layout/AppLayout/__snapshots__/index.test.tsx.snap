// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppLayout > renders correctly and matches snapshot 1`] = `
<div>
  <main
    class="max-w-screen-xl mx-auto"
  >
    <figure>
      NavigationHeader
    </figure>
    <figure
      data-testid="outlet"
    >
      Outlet Component
    </figure>
  </main>
</div>
`;
